src/Rings/PBWAlgebra.jl

```diff
@@ -1171,13 +1171,67 @@ end
 @doc Markdown.doc"""
     eliminate(I::PBWAlgIdeal, V::Vector{<:PBWAlgElem}; ordering = nothing)
 
-Given a vector `V` variables, these variables are eliminated from `I`.
-That is, return the ideal of all elements of `I` which only depend on the remaining variables.
+Given a vector `V` of variables, these variables are eliminated from `I`.
+That is, return the ideal generated by all polynomials in `I` which only involve the remaining variables.
+
+    eliminate(I::PBWAlgIdeal, V::Vector{Int}; ordering = nothing)
+
+Given a vector `V` of indices which specify variables, these variables are eliminated from `I`.
+That is, return the ideal generated by all polynomials in `I` which only involve the remaining variables.
+
 
 !!! note
-    If provided, the `ordering` must be an admissible elimination ordering.
-    If `ordering` is not provided, the function may throw an error if an
-    appropriate ordering could not be found.
+    The return value is an ideal of the original algebra.
+
+!!! note
+    If provided, the `ordering` must be an admissible elimination ordering (this is checked by the function).   
+    If not provided, finding an admissible elimination ordering may involve solving a particular
+    linear programming problem. Here, the function is implemented so that
+    it searches for solutions in a certain range only. If no solution is found
+    in that range, the function will throw an error.
+
+# Examples
+```jldoctest
+julia> R, (x, y, z, a) = QQ["x", "y", "z", "a"];
+
+julia> L = [x*y-z, x*z+2*x, x*a, y*z-2*y, y*a, z*a];
+
+julia> REL = strictly_upper_triangular_matrix(L);
+
+julia> A, (x, y, z, a) = pbw_algebra(R, REL, deglex(gens(R)))
+(PBW-algebra over Rational Field in x, y, z, a with relations y*x = x*y - z, z*x = x*z + 2*x, a*x = x*a, z*y = y*z - 2*y, a*y = y*a, a*z = z*a, PBWAlgElem{fmpq, Singular.n_Q}[x, y, z, a])
+
+julia> f = 4*x*y+z^2-2*z-a;
+
+julia> I = left_ideal(A, [x^2, y^2, z^2-1, f])
+left_ideal(x^2, y^2, z^2 - 1, 4*x*y + z^2 - 2*z - a)
+
+julia> eliminate(I, [x, y, z])
+left_ideal(a - 3)
+
+julia> eliminate(I, [1, 2 ,3])
+left_ideal(a - 3)
+
+julia> try eliminate(I, [z, a]); catch e; e; end
+ErrorException("no elimination is possible: subalgebra is not admissible")
+```
+
+```jldoctest
+julia> R, (p, q) = QQ["p", "q"];
+
+julia> L = [p*q+q^2];
+
+julia> REL = strictly_upper_triangular_matrix(L);
+
+julia> A, (p, q) = pbw_algebra(R, REL, lex(gens(R)))
+(PBW-algebra over Rational Field in p, q with relations q*p = p*q + q^2, PBWAlgElem{fmpq, Singular.n_Q}[p, q])
+
+julia> I = left_ideal(A, [p, q])
+left_ideal(p, q)
+
+julia> try eliminate(I, [q]); catch e; e; end
```

Comment:
```suggestion
julia> try eliminate(I, [q]); catch e; e; end        # in fact, no elimination ordering exists
```
so we don't look bad?